module/documents/activity/mixin.mjs

```diff
@@ -47,6 +78,548 @@ export default Base => class extends PseudoDocumentMixin(Base) {
+    }
+    if ( data.materials?.value ) {
+      supplements.push(`<strong>${game.i18n.localize("DND5E.Materials")}</strong> ${data.materials.value}`);
+    }
+    return {
+      activity: this,
+      actor: this.item.actor,
+      item: this.item,
+      token: this.item.actor?.token,
+      buttons: null,
+      description: data.description.chat,
+      properties: properties.length ? properties : null,
+      subtitle: data.subtitle ?? this.item.system.chatFlavor,
+      supplements
+    };
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Display a chat message for this usage.
+   * @param {ActivityMessageConfiguration} message  Configuration info for the created message.
+   * @returns {Promise<ChatMessage5e|ActivityMessageConfiguration>}
+   * @protected
+   */
+  async _createUsageMessage(message) {
+    const context = await this._usageChatContext();
+    const messageConfig = foundry.utils.mergeObject({
+      rollMode: game.settings.get("core", "rollMode"),
+      data: {
+        content: await renderTemplate(this.metadata.usage.chatCard, context),
+        speaker: ChatMessage.getSpeaker({ actor: this.item.actor }),
+        flags: {
+          core: { canPopout: true }
+        }
+      }
+    }, message);
+
+    /**
+     * A hook event that fires before an activity usage card is created.
+     * @function dnd5e.preCreateUsageMessage
+     * @memberof hookEvents
+     * @param {Activity} activity                     Activity for which the card will be created.
+     * @param {ActivityMessageConfiguration} message  Configuration info for the created message.
+     */
+    Hooks.callAll("dnd5e.preCreateUsageMessage", this, messageConfig);
+
+    ChatMessage.applyRollMode(messageConfig.data, messageConfig.rollMode);
+    const card = messageConfig.create === false ? messageConfig : await ChatMessage.create(messageConfig.data);
+
+    /**
+     * A hook event that fires after an activity usage card is created.
+     * @function dnd5e.postCreateUsageMessage
+     * @memberof hookEvents
+     * @param {Activity} activity                                Activity for which the card was created.
+     * @param {ChatMessage5e|ActivityMessageConfiguration} card  Created card or configuration data if not created.
+     */
+    Hooks.callAll("dnd5e.postCreateUsageMessage", this, card);
+
+    return card;
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Perform any final steps of the activation including creating measured templates.
+   * @param {ActivityUseConfiguration} config  Configuration data for the activation.
+   * @param {ActivityUsageResults} results     Final details on the activation.
+   * @protected
+   */
+  async _finalizeUsage(config, results) {
+    results.templates = [];
+    if ( config.create?.measuredTemplate ) {
+      try {
+        results.templates = await (dnd5e.canvas.AbilityTemplate.fromItem(this.item))?.drawPreview();
```

Comment:
I suppose we need to add a `fromActivity` method at some point? Or are we intending to merge the activity data into the cloned Item as part of the usage workflow? I suppose that could work too.

Reply:
No, I think we'll need to add a new method. I'll stick that in the ToDo list.